From f9f10c3f34fe631d4a3459ebdb5d6525b5e04516 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 16 Jun 2021 23:03:03 +0300
Subject: [PATCH] [SW-46059] gaudi: security: open hbm read only

Once we finish the HBM initialization, we open the whole HBM range to
PCIe for validation.
Unfortunately, the open was done for both: read and write.
The validation scenario requires only read operation, not write.
Fix this by opening for read operations only.

Change-Id: I687d56fd6dff6f5c912e6ce3642d95d5953e4770
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/asic.c b/arch/gaudi/asic.c
index 191e5a5..ff3fa84 100644
--- a/arch/gaudi/asic.c
+++ b/arch/gaudi/asic.c
@@ -266,7 +266,7 @@
 	} else {
 		preboot_set_enabled_cap(HL_FW_EN_CAPS_MEM_CLR);
 		preboot_puts("finished\n");
-		preboot_security_hbm_pci_set_access(true);
+		preboot_security_hbm_pci_open_ro();
 	}
 }
 
diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index ba7ee54..197837b 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -1952,13 +1952,12 @@
 
 #define PCI_RTR_NO	7
 
-void preboot_security_hbm_pci_set_access(bool enable)
+void preboot_security_hbm_pci_open_ro(void)
 {
-	bool is_lock = !enable;
-
 	if (!preboot_security_is_enabled())
 		return;
 
-	preboot_hbw_rr_set_rtr(HBW_RR_HBM, PCI_RTR_NO, HBM_RR_BASE,
-			       is_lock ? HBM_RR_MASK : 0, is_lock);
+	preboot_hbw_rr_set_rtr_wr(HBW_RR_HBM, PCI_RTR_NO, HBM_RR_BASE,
+				  HBM_RR_MASK, true);
+	preboot_hbw_rr_set_rtr_ar(HBW_RR_HBM, PCI_RTR_NO, 0, 0, false);
 }
diff --git a/include/security.h b/include/security.h
index 7b6414b..bbca703 100644
--- a/include/security.h
+++ b/include/security.h
@@ -20,6 +20,6 @@
 void preboot_security_dynamic_img_spsram_open(void);
 void preboot_security_cleanup_before_run(void);
 void preboot_security_sp_sram_write_key_hash(u8 *key, size_t len);
-void preboot_security_hbm_pci_set_access(bool enable);
+void preboot_security_hbm_pci_open_ro(void);
 
 #endif /* _HL_SECURITY_H_ */
